src/core/components/files.js

```diff
@@ -33,16 +35,17 @@ function prepareFile (self, opts, file, callback) {
 
       const b58Hash = cid.toBaseEncodedString()
 
+
       cb(null, {
-        path: file.path || b58Hash,
+        path: opts.wrapWithDirectory ? file.path.substring(WRAPPER.length) : (file.path || b58Hash),
```

Comment:
I admittedly don't know 100% of what's going on here or why the go implementation behaves this way but having it return an empty path for the wrapper directory feels like an anti-pattern.

Reply:
This was explained as "the wrapper is the root"